File: Interop3.cs
﻿using System.Runtime.InteropServices.JavaScript;
using System.Runtime.Versioning;

namespace BlazorWASM70UnmarshalledJS.JavaScriptInterop
{
    [SupportedOSPlatform("browser")]
    public partial class Interop3
    {
        [JSImport("getMessage", "Interop3")]
        internal static partial string GetWelcomeMessage();

        [JSImport("setMessage", "Interop3")]
        internal static partial void SetWelcomeMessage();

        [JSExport]
        internal static string GetMessageFromDotnet()
        {
            return "¡Hola desde Blazor!";
        }
    }
}

File: wwwroot/js/interop3.js
export function getMessage() {
    return '¡Hola desde Blazor!';
}

export async function setMessage() {
    const { getAssemblyExports } = await globalThis.getDotnetRuntime(0);
    var exports = await getAssemblyExports("BlazorWASM70UnmarshalledJS.dll");

    document.getElementById("result").innerText =
        exports.Interop3.GetMessageFromDotnet();
}
